### PR TITLE
NOGW: fix source level calculation and model top momentum conservation

### LINKED ISSUE #1301: Missing: model top treatment for momentum conservation in NOGW
The NOGW needs a momentum conservation treatment when there is flux across the model top

### LINKED ISSUE #1302: NOGW: source level computation
The current source level computation based on source pressure is problematic.

### PR DESCRIPTION
**Pull request by jiahe23**
This PR fixes the gravity wave source level calculation and adds a momentum conservation at model top following MiMA implementation. This brings our results close to MiMA diagnostics.

close #1302 
close #1301

### CODE REVIEW

File: src/tendencies/non_orographic_gravity_wave.jl [hunk 1]
```diff
@@ -326,33 +333,38 @@ function non_orographic_gravity_wave_forcing(
...
+                    (ᶜρ[source_level] / rbh) * fm * eps / (ᶜz[k] - ᶜz[k - 1])
+                if k == length(ᶜu)
+                    wave_forcing[k - 1] = 0.5 * wave_forcing[k - 1]
+                else
+                    wave_forcing[k - 1] =
+                        0.5 * (wave_forcing[k - 1] + wave_forcing[k])
+                end
             else
-                wave_forcing[k] = 0.0
+                wave_forcing[source_level] = 0.0
```
Comment (szy21) [inline src/tendencies/non_orographic_gravity_wave.jl:351]: This line can be moved out of the k loop?

File: test/non_orographic_gravity_wave/nogw_test_mima.jl [hunk 2]
```diff
@@ -197,7 +203,7 @@ gwfu_zonalave = dropdims(mean(gwfu_cgwd, dims = 1), dims = 1)
 gwfv_zonalave = dropdims(mean(gwfv_cgwd, dims = 1), dims = 1)
 
 ENV["GKSwstype"] = "nul"
-output_dir = "nonorographic_gravity_wave_test_mima"
+output_dir = "nonorographic_gravity_wave_test_mima3"
```
Comment (szy21) [inline test/non_orographic_gravity_wave/nogw_test_mima.jl:206]:
```suggestion
output_dir = "nonorographic_gravity_wave_test_mima"
```

**Review by szy21 — APPROVED** [review]
Looks good, thanks!

File: src/tendencies/non_orographic_gravity_wave.jl (hunk 1 above)
Comment (jiahe23) [reply src/tendencies/non_orographic_gravity_wave.jl:351]: Agreed!

**Comment by jiahe23** [comment]
> Looks good, thanks!

Thanks! 

We might need to change the model top (remove it lol). And this will cause discrepancies when compared to mima diagnostics. But at least for now we are confident with the implementation ;)

**Pull request merged**